src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3531,6 +3531,11 @@ public override SyntaxNode ExpressionStatement(SyntaxNode expression)
             return SyntaxFactory.ExpressionStatement((ExpressionSyntax)expression);
         }
 
+        public override SyntaxNode ConditionalAccessExpression(SyntaxNode expression, SyntaxNode whenNotNull)
+        {
+            return SyntaxFactory.ConditionalAccessExpression(Parenthesize(expression), SyntaxFactory.MemberBindingExpression((SimpleNameSyntax)whenNotNull));
```

Comment:
That's not what is causing this issue, this happens after removing my
parenthesizing. It's an existing method that's adding them.
Op 21 jun. 2016 02:27 schreef "CyrusNajmabadi" <notifications@github.com>:

> In src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs
> <https://github.com/dotnet/roslyn/pull/11647#discussion_r67790671>:
>
> > @@ -3531,6 +3531,11 @@ public override SyntaxNode ExpressionStatement(SyntaxNode expression)
> >              return SyntaxFactory.ExpressionStatement((ExpressionSyntax)expression);
> >          }
> >
> > +        public override SyntaxNode ConditionalAccessExpression(SyntaxNode expression, SyntaxNode whenNotNull)
> > +        {
> > +            return SyntaxFactory.ConditionalAccessExpression(Parenthesize(expression), SyntaxFactory.MemberBindingExpression((SimpleNameSyntax)whenNotNull));
>
> Yes. Your code generation logic is wrong :) As kevin mentioned, you should
> not be parenthesizing the whenNotNull portion.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/roslyn/pull/11647/files/717e2490e2a35559b6db7f2ff6b05e0f83b9ce52#r67790671>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ2JlKF8va4z2BV301bJCpCpbJNlvxitks5qNy_sgaJpZM4IqIKZ>
> .
>